api_views/books.py

```diff
@@ -77,4 +77,85 @@ def get_by_title(book_title):
                 }
                 return Response(json.dumps(responseObject), 200, mimetype="application/json")
             else:
-                return Response(error_message_helper("Book not found!"), 404, mimetype="application/json")
\ No newline at end of file
+                return Response(error_message_helper("Book not found!"), 404, mimetype="application/json")
+
+def update_book(book_title):
+    request_data = request.get_json()
+    resp = token_validator(request.headers.get('Authorization'))
+    if "expired" in resp:
+        return Response(error_message_helper(resp), 401, mimetype="application/json")
+    elif "Invalid token" in resp:
+        return Response(error_message_helper(resp), 401, mimetype="application/json")
+    else:
+        user = User.query.filter_by(username=resp).first()
+        book = Book.query.filter_by(user=user, book_title=str(book_title)).first()
+        
+        if book:
+            # Update book fields if present in request
+            if 'book_title' in request_data:
+                book.book_title = request_data['book_title']
+            if 'secret' in request_data:
+                book.secret_content = request_data['secret']
+            
+            db.session.commit()
+            responseObject = {
+                'status': 'success',
+                'message': 'Book has been updated.'
+            }
+            return Response(json.dumps(responseObject), 200, mimetype="application/json")
+        else:
+            return Response(error_message_helper("Book not found!"), 404, mimetype="application/json")
+
+
+def delete_book(book_title):
+    resp = token_validator(request.headers.get('Authorization'))
+    if "expired" in resp:
+        return Response(error_message_helper(resp), 401, mimetype="application/json")
+    elif "Invalid token" in resp:
+        return Response(error_message_helper(resp), 401, mimetype="application/json")
+    else:
+        user = User.query.filter_by(username=resp).first()
+        book = Book.query.filter_by(user=user, book_title=str(book_title)).first()
+        
+        if book:
+            db.session.delete(book)
+            db.session.commit()
+            responseObject = {
+                'status': 'success',
+                'message': 'Book has been deleted.'
+            }
+            return Response(json.dumps(responseObject), 200, mimetype="application/json")
+        else:
+            return Response(error_message_helper("Book not found!"), 404, mimetype="application/json")
+
+
+def get_books_by_user(username):
```

Comment:
get_books_by_user function should validate if the requesting user has permission to view other users' books